File: Sources/MagicDiffView/DemoFiles/go/user_model.before.go
package main

import "fmt"

type User struct {
	ID    int
	Name  string
	Email string
}

func (u *User) Validate() bool {
	return u.Name != "" && u.Email != ""
}

func (u *User) String() string {
	return fmt.Sprintf("User(%d): %s <%s>", u.ID, u.Name, u.Email)
}

type UserManager struct {
	users []User
}

func (m *UserManager) AddUser(user User) {
	m.users = append(m.users, user)
}

func (m *UserManager) FindUser(id int) *User {
	for i := range m.users {
		if m.users[i].ID == id {
			return &m.users[i]
		}
	}
	return nil
}

func (m *UserManager) RemoveUser(id int) bool {
	for i, user := range m.users {
		if user.ID == id {
			m.users = append(m.users[:i], m.users[i+1:]...)
			return true
		}
	}
	return false
}

func (m *UserManager) TotalCount() int {
	return len(m.users)
}

File: Sources/MagicDiffView/DemoFiles/rust/user_model.before.rs
use std::fmt;

#[derive(Debug, Clone)]
pub struct User {
    pub id: i32,
    pub name: String,
    pub email: String,
}

impl User {
    pub fn new(id: i32, name: String, email: String) -> Self {
        User { id, name, email }
    }

    pub fn validate(&self) -> bool {
        !self.name.is_empty() && !self.email.is_empty()
    }
}

impl fmt::Display for User {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "User({}): {} <{}>", self.id, self.name, self.email)
    }
}

pub struct UserManager {
    users: Vec<User>,
}

impl UserManager {
    pub fn new() -> Self {
        UserManager { users: Vec::new() }
    }

    pub fn add_user(&mut self, user: User) {
        self.users.push(user);
    }

    pub fn find_user(&self, id: i32) -> Option<&User> {
        self.users.iter().find(|u| u.id == id)
    }

    pub fn remove_user(&mut self, id: i32) -> bool {
        let initial_len = self.users.len();
        self.users.retain(|u| u.id != id);
        self.users.len() < initial_len
    }

    pub fn total_count(&self) -> usize {
        self.users.len()
    }
}

impl Default for UserManager {
    fn default() -> Self {
        Self::new()
    }
}
